src/App.js

```diff
@@ -17,10 +18,27 @@ function App() {
         <Route path="/ProductList" element={<ProductList />}>
           ProductList
         </Route>
-        <Route path="/WishList" element={<WishList />}>
+        <Route path="/UserInfo" element={<UserInfo />}>
+          UserInfo
+        </Route>
+        <Route
+          path="/WishList"
+          element={
+            <RequiresAuth>
+              <WishList />
+            </RequiresAuth>
```

Comment:
you can create nested routes where you need to run the RequiresAuth only once,
read about auth with outlet from react-router
For now its fine

Reply:
Sure. I will look into that 👍🏻